src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -579,28 +589,65 @@ private static String getStringInsideTag(String result, DetailNode detailNode) {
      * Finds and returns first sentence.
      *
      * @param ast Javadoc root node.
+     * @param period Period character.
      * @return first sentence.
      */
-    private static String getFirstSentence(DetailNode ast) {
+    private static String getFirstSentence(DetailNode ast, String period) {
+        final Deque<DetailNode> stack = new LinkedList<>();
```

Comment:
I've named it nodesToProcess. It's always awkward naming a stack because the type of the variable doesn't tell you that you're just using it as a stack.